pkgs/native_assets_builder/test/build_runner/build_runner_caching_test.dart

```diff
@@ -22,6 +22,10 @@ void main() async {
         workingDirectory: packageUri,
         logger: logger,
       );
+      // Make sure the first compile is at least one second after the
+      // package_config.json is written, otherwise dill compilation isn't
+      // cached.
+      await Future<void>.delayed(const Duration(seconds: 1));
```

Comment:
Using [`Clock`](https://pub.dev/documentation/clock/latest/clock/Clock-class.html) could probably solve this without introducing delays, but is maybe not necessary if it happens only here.

Reply:
Yeah, it's just the test.

In real use cases it will probably be slower. And if it's as fast as this, we don't mind rerunning it.